storage/shard.go

```diff
@@ -50,6 +61,41 @@ func (s *dbShard) Write(ev event.Event) error {
 	return segment.Write(ev)
 }
 
+// TODO(xichen): retry logic on segment persistence failure.
+func (s *dbShard) Flush(ps persist.Persister) error {
+	var multiErr xerrors.MultiError
+	s.Lock()
+	activeSegment := s.active
+	s.active = newDatabaseSegment(s.opts)
```

Comment:
During the flush phase, it looks like `activeSegment` is invisible to the shard. We won't be able to query it?

Reply:
Honestly havent' given much thought to the read API at this point, was thinking either you or I'll make whatever changes necessary when implementing the read API.